### PR TITLE
#1203 slider improving label positioning and width

### PR DESCRIPTION
**Pull request by ktabors**
Fixes a few issues in https://github.com/adobe/react-spectrum/issues/1203

Moves the label value to the edge when on top and align start when a side label.
Default width is 192px.
Changed the width on the width and label overflow stories.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Test all the slider stories, specifically the value stories, the the label side, the label overflow, and the custom width stories. 

## 🧢 Your Project:
RSP

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/slider/index.css [hunk 1]
```diff
@@ -66,10 +66,10 @@ governing permissions and limitations under the License.
   z-index: auto;
 
   min-block-size: var(--spectrum-slider-height);
-  min-inline-size: var(--spectrum-slider-min-width);
+  min-inline-size: var(--spectrum-global-dimension-size-2400);
```
Comment (ktabors) [inline packages/@adobe/spectrum-css-temp/components/slider/index.css:69]: Should I create a PR to adjust the min-width to 192px here? https://git.corp.adobe.com/Spectrum/spectrum-dna/blob/main/data/elements/slider/SliderBase.mjs#L40

Comment (LFDanLu) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:69]: Perhaps a variable in the `:root` of this file would suffice for now? Something like `--spectrum-slider-min-height: var(--spectrum-global-dimension-size-2400);`

Comment (LFDanLu) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:69]: Ah never mind, my mistake. Just realized that `--spectrum-slider-min-width` is already a thing

**Review by LFDanLu — COMMENTED** [review]
Is the default slider width of 192px supposed to be applied to outer slider div (aka `spectrum-Slider`) or the slider control (aka `spectrum-Slider-controls`)? Been trying to figure out which makes sense but the XD file and spectrum-css docs don't show anything being 192px by default

The answer to the above will affect things like the inline-size calculation the slider-control does and stuff

**Comment by ktabors** [comment]
Discussed where the default width should be set and it should be on the track. This is how other labeled form components like TextField are done, with it on the input and not the container wrapping the label and the input. The custom width should go on the container.

File: packages/@adobe/spectrum-css-temp/components/slider/index.css (hunk 1 above)
Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:69]: https://git.corp.adobe.com/Spectrum/spectrum-dna/pull/534

Comment (devongovett) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:69]: Is it supposed to change the min width or just the default width?

File: packages/@adobe/spectrum-css-temp/components/slider/index.css [hunk 2]
```diff
@@ -323,6 +324,10 @@ governing permissions and limitations under the License.
   cursor: default;
   font-feature-settings: "tnum";
   text-align: end;
+
+  &:only-child {
```
Comment (devongovett) [inline packages/@adobe/spectrum-css-temp/components/slider/index.css:328]: What's the `:only-child` for?

File: packages/@react-spectrum/slider/src/RangeSlider.tsx [hunk 3]
```diff
@@ -60,7 +60,7 @@ function RangeSlider(props: SpectrumRangeSliderProps, ref: FocusableRef<HTMLDivE
         let upperTrack = (
           <div
             className={classNames(styles, 'spectrum-Slider-track')}
-            style={{[cssDirection]: `${state.getThumbPercent(1) * 100}%`, width: `${(1 - state.getThumbPercent(1)) * 100}%`}} />
```
Comment (devongovett) [inline packages/@react-spectrum/slider/src/RangeSlider.tsx:63]: What was this change for?

Comment (LFDanLu) [reply packages/@react-spectrum/slider/src/RangeSlider.tsx:63]: Ah this one I can answer: @ktabors had noticed the following issue with range sliders where the right most track line spills into the right thumb:
image
Looking at https://opensource.adobe.com/spectrum-css/slider.html#range, the right most track line only has width applied to it, the `left`/`right` inline style here was extraneous

File: packages/@adobe/spectrum-css-temp/components/slider/index.css (hunk 2 above)
Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:328]: When it is a side label, the value and label are in separate label containers... the way I added the start align to the value when side label is that value is the only child of a label container.

Comment (devongovett) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:328]: Oh I assumed there was a class for the side label state but I guess not... 😞 

Maybe just add a comment to explain what you just said then.

File: packages/@adobe/spectrum-css-temp/components/slider/index.css (hunk 1 above)
Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:69]: min width is default width, right?

File: packages/@adobe/spectrum-css-temp/components/slider/index.css [hunk 4]
```diff
@@ -69,7 +75,7 @@ governing permissions and limitations under the License.
   min-inline-size: var(--spectrum-slider-min-width);
 
   /* These calculations prevent the track from spilling outside of the control */
-  inline-size: calc(100% - calc(var(--spectrum-slider-controls-margin) * 2));
+  inline-size: calc(100% - calc(var(--spectrum-slider-controls-margin)));
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/slider/index.css:78]: Since the default width is being specified on the wrapping div now, I _think_ you an revert this change here and remove the `margin-inline-end` that you added below

Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:103]: danni/devon removed it

File: packages/@adobe/spectrum-css-temp/components/slider/index.css [hunk 5]
```diff
@@ -58,9 +58,35 @@ governing permissions and limitations under the License.
   touch-action: none;
 
   width: var(--spectrum-global-dimension-size-2400);
+  min-inline-size: var(--spectrum-slider-min-width);
```
Comment (ktabors) [inline packages/@adobe/spectrum-css-temp/components/slider/index.css:61]: I thought textfield had the min width on the input parent not the field and input parent parent.

File: packages/@adobe/spectrum-css-temp/components/slider/index.css (hunk 2 above)
Comment (devongovett) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:358]: Maybe we should use the `spectrum-Slider--positionSide` class now?

File: packages/@adobe/spectrum-css-temp/components/slider/index.css (hunk 5 above)
Comment (dannify) [reply packages/@adobe/spectrum-css-temp/components/slider/index.css:61]: Yeah honestly I don't know which is right. I am going to bring it up with Spectrum because I don't think I was specific enough when I asked them originally.

File: packages/@adobe/spectrum-css-temp/components/slider/index.css [hunk 6]
```diff
@@ -56,6 +56,43 @@ governing permissions and limitations under the License.
...
+
+    .spectrum-Slider-controls {
+      flex: 1;
+      width: var(--spectrum-component-single-line-width);
+    }
+    .spectrum-Slider-labelContainer {
+      margin-inline-end: calc(var(--spectrum-slider-label-gap-x) / 2);
+    }
+
+    &:only-child {
```
Comment (devongovett) [inline packages/@adobe/spectrum-css-temp/components/slider/index.css:91]:
```suggestion
    .spectrum-Slider-value {
```
